mesonbuild/dependencies/base.py

```diff
@@ -929,31 +937,69 @@ def __init__(self, name, environment, kwargs, language=None):
+        def search():
+            # Lookup in cross or machine file.
+            potential_cmake = environment.binaries[for_machine] \
+                .lookup_entry('cmake')
+            if potential_cmake is not None:
+                mlog.debug('CMake binary for %s specified from cross file, native file, or env var as %s.', for_machine, potential_cmakepath)
+                yield ExternalProgram.from_entry('cmake', potential_cmake)
+                # We never fallback if the user-specified option is no good, so
+                # stop returning options.
+                return
+            mlog.debug('CMake binary missing from cross or native file, or env var undefined.')
+            # Fallback on hard-coded defaults.
+            # TODO prefix this for the cross case instead of ignoring thing.
+            if environment.machines.matches_build_machine(for_machine):
+                for potential_cmakepath in environment.default_cmake:
+                    mlog.debug('Trying a default CMake fallback at', potential_cmake)
+                    yield ExternalProgram(potential_cmakepath, silent=True)
+
+        # Only search for CMake the first time and store the result in the class
+        # definition
+        if CMakeDependency.class_cmakebin[for_machine] is False:
+            mlog.debug('CMake binary for %s is cached missing.' % for_machine)
+        elif CMakeDependency.class_cmakebin[for_machine] is not None:
+            mlog.debug('CMake binary for %s is cached.' % for_machine)
+        else:
+            assert CMakeDependency.class_cmakebin[for_machine] is None
+            mlog.debug('CMake binary for %s is not cached.', for_machine)
+            for potential_cmakebin in search():
+                mlog.debug(
+                    'Trying CMake binary %s for machine %s at %s.',
+                    potential_cmakebin.name, for_machine, potential_cmakebin.command)
+                version_if_ok = self.check_cmake(potential_cmakebin)
+                if not version_if_ok:
+                    continue
+                if not self.silent:
+                    mlog.log('Found CMake:', mlog.bold(potential_cmakebin.get_path()),
+                             '(%s)' % version_if_ok)
+                CMakeDependency.class_cmakebin[for_machine] = potential_cmakebin
+                CMakeDependency.class_cmakevers[for_machine] = version_if_ok
+                break
+            else:
+                if not self.silent:
+                    mlog.log('Found CMake:', mlog.red('NO'))
+                # Set to False instead of None to signify that we've already
+                # searched for it and not found it
+                CMakeDependency.class_cmakebin[for_machine] = False
+                CMakeDependency.class_cmakebin[for_machine] = None
+
+        self.cmakebin = CMakeDependency.class_cmakebin[for_machine]
+        self.cmakevers = CMakeDependency.class_cmakevers[for_machine]
+        if self.cmakebin == False:
```

Comment:
__[Flake8]__ [E712] comparison to False should be 'if cond is False:' or 'if not cond:'

<sub>posted by Sider</sub>